### PR TITLE
WT-6383 Add mixed mode validation

### PR DESCRIPTION
**Pull request by tetsuo-cpp**
This PR is adding another type of key validation called `mixed_mode` as an alternative to the `key_consistent` validation we offer.

This setting triggers errors for out-of-order timestamps but allows the interleaving of mixed timestamped and non-timestamped updates. Non-timestamped updates are considered to reset the chain so we can have:

```
10 => 0 => 5
```
I don't think Mongo does this but it's easier to implement like this and fits better with how we expect mixed-mode to work in general.

### CODE REVIEW

File: src/txn/txn.c [hunk 1]
```diff
@@ -1277,7 +1287,8 @@ __txn_commit_timestamps_assert(WT_SESSION_IMPL *session)
             op_timestamp = txn->commit_timestamp;
         if (F_ISSET(txn, WT_TXN_TS_COMMIT_KEYS) && op_timestamp < prev_op_timestamp)
```
Comment (quchenhao) [inline src/txn/txn.c:1288]: So we allow commit timestamp to be out of order?

Comment (quchenhao) [reply src/txn/txn.c:1288]: In history store, the keys are ordered by commit timestamp not durable timestamp.

Comment (tetsuo-cpp) [reply src/txn/txn.c:1288]: They've asked for a `durable_timestamp=mixed_mode` setting.

Comment (tetsuo-cpp) [reply src/txn/txn.c:1288]: I'll check whether that's what they actually wanted.

**Comment by tetsuo-cpp** [comment]
This is blocked by #6121.
I'll help get that one merged and then come back to this ticket later.

**Pull request closed without merging**